src/libcore/fmt/mod.rs

```diff
@@ -81,6 +81,21 @@ pub type Result = result::Result<(), Error>;
 /// This type does not support transmission of an error other than that an error
 /// occurred. Any extra information must be arranged to be transmitted through
 /// some other means.
+///
+/// The key thing to remember here is that the type `fmt::Error` should not be
+/// confused with `std::error::Error`, which you may also have in scope.
```

Comment:
If we're comparing to other `Error`s within the standard library, it might also be good to include `std::io::Error`, which I would consider more well-known than the Error trait.